### DISCUSSION TITLE
How can I squash the `stuck thread` warning from XLA?

### DISCUSSION THREAD

**Discussion by C-J-Cundy** [jax-ml/jax#13305] [Q&A]
I'm doing a task where I'm pmapping an operation that takes a variable amount of time over different devices. 

This means that quite often, one device will finish much faster than the others. When this happens, I get this error message: 
```
022-11-17 16:25:04.454825: E external/org_tensorflow/tensorflow/compiler/xla/service/rendezvous.cc:31] This thread has been waiting for 10 seconds and may be stuck:

2022-11-17 16:25:15.673476: E external/org_tensorflow/tensorflow/compiler/xla/service/rendezvous.cc:36] Thread is unstuck! Warning above was a false-positive. Perhaps the timeout is too short.
```

Is there a way to increase the timeout to stop this error message being printed, perhaps with an XLA flag? 
It really clutters up my stdout when training. 
I would prefer to avoid a solution that involves suppressing all error messages, as they can be very useful.

**Comment by bjuncek** [comment]
ping on this?

**Comment by bmorris3** [comment]
I would like this too.

**Comment by olupton** [comment]
The pasted log doesn't show which rendezvous is causing the warning, but you could try `--xla_gpu_executable_warn_stuck_timeout_seconds=XXX` (which does default to 10) and `--xla_gpu_executable_terminate_timeout_seconds=YYY` (default 30) in `XLA_FLAGS`.
